src/ghci/stdin.rs

```diff
@@ -135,15 +137,64 @@ impl GhciStdin {
     }
 
     #[instrument(skip(self, stdout), level = "debug")]
-    pub async fn show_modules(&mut self, stdout: &mut GhciStdout) -> miette::Result<ModuleSet> {
+    pub async fn show_paths(&mut self, stdout: &mut GhciStdout) -> miette::Result<ShowPaths> {
         self.set_mode(stdout, Mode::Internal).await?;
 
         self.stdin
-            .write_all(b":show modules\n")
+            .write_all(b":show paths\n")
             .await
             .into_diagnostic()?;
 
-        stdout.show_modules().await
+        stdout.show_paths().await
+    }
+
+    #[instrument(skip(self, stdout), level = "debug")]
+    pub async fn show_targets(
+        &mut self,
+        stdout: &mut GhciStdout,
+        show_paths: &ShowPaths,
+    ) -> miette::Result<ModuleSet> {
+        self.set_mode(stdout, Mode::Internal).await?;
```

Comment:
as a side note, i think at some point (doesn't have to be in this PR) the mode system should be changed so that the mode is provided as an argument to write commands instead of being set out-of-band and tracked as a mutable variable.   That would prevent quite a few potential bugs due to the wrong mode being set for a given write

Reply:
I agree with setting the mode as part of the write commands, but I think it'll still have to be tracked as a mutable variable because the mode influences the way the stdout and stderr readers process their output. (They need to be able to know what the last mode was, not just when a write happens.)